napari/utils/events/event.py

```diff
@@ -228,13 +233,20 @@ class EventEmitter:
         The class of events that this emitter will generate.
     """
 
-    def __init__(self, source=None, type=None, event_class=Event):
-        self._callbacks = []
-        self._callback_refs = []
+    def __init__(
+        self,
+        source: Any = None,
+        type: str = None,
+        event_class: Type[Event] = Event,
+    ):
+        # connected callbacks
+        self._callbacks: list[Callback | CallbackRef] = []
+        # used when connecting new callbacks at specific positions
+        self._callback_refs: list[str | None] = []
```

Comment:
The `|` syntax is supposed to be only 3.10+, I believe  [pep 604](https://www.python.org/dev/peps/pep-0604/) it will import properly as you have `from __future__ import annotation`; but may not do the right things when using `get_type_hints()`. I'm fine using it; it may affect Jedi and other completer though, not sure.

Reply:
Those changes were originally suggested by [pyupgrade](https://github.com/asottile/pyupgrade), which we use as a commit hook.  I've since been using it and testing it pretty regularly back to 3.7 on `magicgui`... and haven't seen any issues.  but it's a good point about `get_type_hints`.  I haven't specifically checked that

Reply:
I guess that's why pyupgrade have the --keep-runtime-typomg option. It's likely a detail and not too important, but want to make sure decisions are taken into consideration.

Reply:
I decided to undo these `__future__` additions.  It would be the first place we added them, so let's punt on that.